### PR TITLE
Update flow parser to 0.84.0

### LINKED ISSUE #5084: "export default async function" wraps function in parenthesis
I know there's already an issue for it - https://github.com/prettier/prettier/issues/700
but it's closed and yet not fixed.
Using the prettier version 1.14.0 - I still get this error (while using flow parser).

### PR DESCRIPTION
**Pull request by jbrown215**
Fixes #5084

Update the flow parser to 0.84.0, which includes parser support for explicit inexact objects using `...`. See https://medium.com/flow-type/on-the-roadmap-exact-objects-by-default-16b72933c5cf for more info.

One test had to be re-recorded

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: tests/es6modules/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -80,7 +80,7 @@ export default async function foo() {}
 exports[`export_default_function_declaration_async.js - flow-verify 1`] = `
 export default async function foo() {}
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-export default (async function foo() {});
+export default async function foo() {}
```
Comment (lydell) [inline tests/es6modules/__snapshots__/jsfmt.spec.js.snap]: Do we know why this happens? Is it valid?

Comment (jbrown215) [reply tests/es6modules/__snapshots__/jsfmt.spec.js.snap]: No idea, but it is valid flow syntax: https://flow.org/try/#0KYDwDg9gTgLgBAE2AMwIYFcA29UGcCeAdgMZzLokwCWEhZEEAFAJRwDeAvkA

Comment (ikatyang) [reply tests/es6modules/__snapshots__/jsfmt.spec.js.snap]: See https://github.com/prettier/prettier/issues/5084#issuecomment-420489740.

**Comment by ikatyang** [comment]
Thanks!

**Pull request merged**